Neos.Neos/Classes/Domain/Repository/WorkspaceMetadataAndRoleRepository.php

```diff
@@ -319,6 +353,8 @@ public function findPrimaryWorkspaceNameForUser(ContentRepositoryId $contentRepo
                 AND classification = :personalWorkspaceClassification
                 AND owner_user_id = :userId
         SQL;
+        // We don't order the results and return the first matching workspace.
+        // In case multiple exist - which is not desired currently - and in the happy path not possible via the api (race conditions aside) - the order is not defined.
```

Comment:
just wondering: couldn't this enforced on the db level?

Reply:
actually yes good point i thought at first no, but an unique index on `owner_user_id` AND `content_repository_id` would be possible and makes this super easy. And if we dont need that later we could drop that index.
Will do that next commit!

Reply:
Implemented via [TASK: Ensure that the workspace owner is unique via sql index](https://github.com/neos/neos-development-collection/pull/5334/commits/ddccba16f5168bd6c97b439ee729df253c68330a)